### PR TITLE
DG update with implementation details of Start/End command

### PR DESCRIPTION
**Pull request by WangCheng0116**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ The `UI` component,
 
 ### Logic component
 
-**API** : [`Logic.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/logic/Logic.java)
+**API** : [`Logic.java`](https://github.com/AY2324S1-CS2103T-T11-4/tp/blob/master/src/main/java/seedu/flashlingo/logic/Logic.java)
```
Comment (A1WAYSD) [inline docs/DeveloperGuide.md:84]: Thanks for adapting this to our own project!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -148,91 +148,101 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+maintaining the state of whether the session is a review session or not. With the Singleton pattern in place, you can be
+confident that there is only one `SessionManager` instance, making it a centralized and controlled entity for session 
+management within the application.
 
-Step 3. The user executes `add n/David …​` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+**Step 3:** The user executes various commands within the action sequence, such as `yes` and `no`.
 
-![UndoRedoState2](images/UndoRedoState2.png)
+**Step 4:** The user chooses to end the review session by using `end` command. This action will alternate the boolean value 
+inside SessionManager class indicating current session is review session or not.
```
Comment (A1WAYSD) [inline docs/DeveloperGuide.md:181]: Good job! Detailed explanation of sessionManager.

File: src/main/java/seedu/flashlingo/storage/JsonAdaptedFlashCard.java [hunk 3]
```diff
@@ -102,9 +102,6 @@ public FlashCard toModelType() throws IllegalValueException {
         }
         final Date modelWhenToReview = Date.from(ZonedDateTime.parse(whenToReview).toInstant());
 
-        if (!ProficiencyLevel.isValidProficiencyLevel(level)) {
-            throw new IllegalValueException(ProficiencyLevel.MESSAGE_CONSTRAINTS);
-        }
```
Comment (A1WAYSD) [inline src/main/java/seedu/flashlingo/storage/JsonAdaptedFlashCard.java:107]: Is it safe to completely delete this? What if the level is negative.

Comment (WangCheng0116) [reply src/main/java/seedu/flashlingo/storage/JsonAdaptedFlashCard.java]: I have put this checking back. Things should be fine now.

**Review by Song-Mengfei — APPROVED** [review]
Good

**Pull request merged**